install/tk.sh

```diff
@@ -1,50 +1,83 @@
 #!/usr/bin/env bash
-
 set -e
 
 DEFAULT_BIN_DIR="/usr/local/bin"
 BIN_DIR=${1:-"$DEFAULT_BIN_DIR"}
 
-opsys=""
-if [[ "$OSTYPE" == linux* ]]; then
-  opsys=linux
-elif [[ "$OSTYPE" == darwin* ]]; then
-  opsys=darwin
+# Determine OS and CPU architecture
+if [[ $(uname) = Darwin ]]; then
+    OS=darwin
+elif [[ $(uname) = Linux ]]; then
+    OS=linux
+else
+    OS=unknown
 fi
 
-if [[ "$opsys" == "" ]]; then
-  echo "OS $OSTYPE not supported"
-  exit 1
+case $(uname -m) in
+x86_64)
+    ARCH=amd64
+    ;;
+x86|i386|i686)
+    ARCH=386
+    ;;
+arm|armv7*)
+    ARCH=armv6
+    ;;
+arm64|aarch64*|armv8*)
+    ARCH=arm64
+    ;;
+*)
+    ARCH=unknown
+    ;;
+esac
+
+if [[ $OS = unknown ]] || [[ $ARCH = unknown ]]; then
+    echo "Unable to detect your OS and CPU architecture."
+    exit 1
+fi
+if [[ $ARCH != amd64 ]]; then
+    echo "CPU architecture $ARCH is not supported."
+    exit 1
 fi
 
 if [[ ! -x "$(command -v curl)" ]]; then
```

Comment:
Another improvement here may be to fallback to `wget` as either `curl` or `wget` is available on most system.

Reply:
I rewrote the script and added support for a downloader fallback.